perfmetrics/scripts/fio/fio_metrics.py

```diff
@@ -43,12 +44,130 @@
 P50 = '50.000000'
 P90 = '90.000000'
 P95 = '95.000000'
-LAT_20_PERC = 'lat_20_perc'
-LAT_50_PERC = 'lat_50_perc'
-LAT_90_PERC = 'lat_90_perc'
-LAT_95_PERC = 'lat_95_perc'
 IO_BYTES = 'io_bytes'
 
+NAME = 'name'
+JSON_NAME = 'json_name'
+FORMAT = 'format'
+DEFAULT = 'default'
+LEVELS = 'levels'
+CONVERSION = 'conversion'
+NS_TO_S = 10**(-9)
+
+""" REQ_JOB_PARAMS:
+NAME: key for the parameter, to be used when creating parameter dict for each
+Job
+JSON_NAME: Key for parameter inside 'global options'/'job options' dictionary
+  Ex: For output json = {"global options": {"filesize":"50M"}, "jobs": [
+  "job options": {"rw": "read"}]}
+  JSON_NAME for filesize will be "filesize" and that for readwrite will be "rw"
+FORMAT: Function returning formatted parameter value. Needed to convert
+parameter to plottable values
+  Ex: 'filesize' is obtained as '50M', but we need to convert it to integer
+  showing size in kb in order to maintain uniformity
+DEFAULT: Default value for the parameter
+
+Ex: output json = {"global options": {"filesize":"50M"}}
+For REQ_JOB_PARAMS = [
+    {
+        NAME: FILESIZE_KB,
+        JSON_NAME: FILESIZE,
+        FORMAT: lambda val: _convert_value(val, FILESIZE_CONVERSION),
+        DEFAULT: 0
+    },
+    {
+        NAME: RW,
+        JSON_NAME: RW,
+        FORMAT: lambda val: val,
+        DEFAULT: 'read'
+    }
+]
+Extracted parameters would be {FILESIZE_KB: 50000, RW: 'read'}
+"""
+REQ_JOB_PARAMS = [
+    {
+        NAME: FILESIZE_KB,
+        JSON_NAME: FILESIZE,
+        FORMAT: lambda val: _convert_value(val, FILESIZE_CONVERSION),
+        DEFAULT: 0
+    },
+    {
+        NAME: THREADS,
+        JSON_NAME: NUMJOBS,
+        FORMAT: lambda val: int(val),
+        DEFAULT: '1'
+    },
+    # Don't remove the below parameter
+    {
+        NAME: RW,
+        JSON_NAME: RW,
+        FORMAT: lambda val: val,
+        DEFAULT: 'read'
+    }
+]
+
+""" REQ_JOB_METRICS:
+NAME: key for the metric, to be used when creating metric dict for each job
+LEVELS: Keys for the metric inside 'read'/'write' dictionary in each job
```

Comment:
same comment as above. levels should match the key in fio job spec

Reply:
done